bundles/org.openhab.core.automation/src/main/java/org/openhab/core/automation/internal/module/handler/GroupCommandTriggerHandler.java

```diff
@@ -59,21 +61,29 @@ public class GroupCommandTriggerHandler extends BaseTriggerModuleHandler impleme
 
     public static final String CFG_GROUPNAME = "groupName";
     public static final String CFG_COMMAND = "command";
+    private final String ruleUID;
 
     private ServiceRegistration<?> eventSubscriberRegistration;
 
-    public GroupCommandTriggerHandler(Trigger module, BundleContext bundleContext, ItemRegistry itemRegistry) {
+    public GroupCommandTriggerHandler(Trigger module, String ruleUID, BundleContext bundleContext,
+            ItemRegistry itemRegistry) {
         super(module);
         this.groupName = (String) module.getConfiguration().get(CFG_GROUPNAME);
         this.command = (String) module.getConfiguration().get(CFG_COMMAND);
-        this.types = Set.of(ItemCommandEvent.TYPE);
+        this.types = Set.of(ItemCommandEvent.TYPE, ItemAddedEvent.TYPE, ItemRemovedEvent.TYPE);
         this.bundleContext = bundleContext;
         this.itemRegistry = itemRegistry;
+        this.ruleUID = ruleUID;
         Dictionary<String, Object> properties = new Hashtable<>();
-        this.topic = "openhab/items/";
+        this.topic = "openhab/items/*";
         properties.put("event.topics", topic);
         eventSubscriberRegistration = this.bundleContext.registerService(EventSubscriber.class.getName(), this,
                 properties);
+
+        if (itemRegistry.get(groupName) == null) {
+            logger.warn("Group '{}' needed for rule '{}', trigger '{}' not present in registry. Trigger will not work.",
```

Comment:
I think the message about the trigger is incorrect/misleading. The trigger is in the registry, the item is not.
I'd suggest to change it to something like, which is anyhow shorter and has all relevant information:
```suggestion
            logger.warn("Group '{}' needed for rule '{}', Trigger '{}' will not work.",
```
Same of course for all the other log messages.

Reply:
Done.